src/store/wallet/wallet-epics.ts

```diff
@@ -166,6 +169,32 @@ const approveInternalOperationRequestEpic = (action$: Observable<Action>, state$
     )
   );
 
+const BCD_INDEXING_DELAY = 15000;
+
+const waitForOperationCompletionEpic = (action$: Observable<Action>) =>
+  action$.pipe(
+    ofType(waitForOperationCompletionAction),
+    toPayload(),
+    withLatestFrom(tezos$),
+    switchMap(([{ opHash, sender }, tezos]) =>
+      from(tezos.operation.createOperation(opHash)).pipe(
+        switchMap(operation => operation.confirmation(1)),
+        switchMap(() => of(null).pipe(delay(BCD_INDEXING_DELAY))),
```

Comment:
no need in `switchMap(() => of(null).pipe()` just use `delay` operator directly

Reply:
```
switchMap(),
delay(),
concatMap()
```

Reply:
Do you really think I wouldn't have used this order if it worked for me?